docs/user/install-gke-external.md

```diff
@@ -0,0 +1,246 @@
+```
+
+Then add the DNS `A` record pointing to the external IP address:
+
+```
+gcloud dns record-sets create <DOMAIN_NAME> --zone="turbinia-dns" --type="A" --ttl="300" --rrdatas="EXTERNAL_IP"
+```
+
+DNS can instead be managed through [ExternalDNS](https://github.com/kubernetes-sigs/external-dns), however setup is outside the scope of this guide.
+
+### 3. Create Oauth2 Application IDs
+
+Authentication is handled by a proxy utility named [Oauth2 Proxy](https://oauth2-proxy.github.io/oauth2-proxy/). This guide will walk through configuring the Oauth2 Proxy with Google Oauth,
+however there are alternative [providers](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) that you may configure instead.
+
+Two sets of Oauth credentials will be configured as part of this deployment.
+One that will be for the Web client and one for the API/Desktop client.
+
+To create the Web Oauth credentials, take the following steps:
+
+1. Go to the [Credentials page](https://console.developers.google.com/apis/credentials).
+2. Click Create credentials > OAuth client ID.
+3. Select the `Web application` application type.
+4. Fill in an appropriate Application name.
+5. Fill in Authorized JavaScript origins with your domain as `https://<DOMAIN_NAME>`
+6. Fill in Authorized redirect URIs with `https://<DOMAIN_NAME>/oauth2/callback`
+7. Please make a note of the generated `Client ID` and `Client Secret` for later use.
+
+To create the API/Desktop Oauth credentials, take the following steps:
+
+1. Go to the [Credentials page](https://console.developers.google.com/apis/credentials).
+2. Click Create credentials > OAuth client ID.
+3. Select the `Desktop or Native application` application type.
+4. Fill in an appropiate application name.
+5. Please make a note of the generated `Client ID` and `Client Secret` for later use.
+
+You will then need to generate a cookie secret for later use:
+
+```
+python3 -c 'import os,base64; print(base64.urlsafe_b64encode(os.urandom(32)).decode())'
+```
+
+With the Turbinia repository cloned to your local machine, cd into the directory we'll be working from:
+
+```
+wyassine@wyassine:~/turbinia$ cd k8s/tools/
+```
+
+Then make a copy of the `oauth2_proxy.cfg` template:
+
+```
+wyassine@wyassine:~/turbinia/k8s/tools$ cp ../../docker/oauth2_proxy/oauth2_proxy.cfg .
+```
+
+Edit the `oauth2_proxy.cfg` file and replace the following:
+
+- `CLIENT_ID`: The web client id
+- `CLIENT_SECRET`: The web client secret
+- `OIDC_EXTRA_AUDIENCES`: The native client id
+- `UPSTREAMS`: The domain name registered, ex: `upstreams = ['https://<DOMAIN>]`
+- `REDIRECT_URL`: The redirect URI you registered ex: `https://<DOMAIN>/oauth2/callback`
+- `COOKIE_SECRET`: The cookie secret you generated above
+- `EMAIl_DOMAINS`: The email domain name you'd allow to authenticate ex: `yourcompany.com`
+
+Now base64 encode the config file:
+
+```
+base64 -w0 oauth2_native.cfg > oauth2_native.b64
```

Comment:
These instructions are great. Thanks for writing this out. In future we can probably ask the user to edit the config file and then have a bash or Python helper script that generates the cookie secret, puts it in the config file and base64 encodes it.

Reply:
Np hopefully can carry over for some of the local setup too, thanks for the review! Not a bad idea for future improvements to automate even further